Bluetooth: controller: Remove BT_HCI_RAW dependency

thomasstenersen opened the pull request:
Removes the dependency on BT_HCI_RAW from BT_CTLR_ASSERT_HANDLER.

For example, when testing the full stack in CI it is useful to re-define the assert handler to get additional system information when an assert is triggered. In this case, BT_HCI_RAW is not necessarily set, since the application is usually running the full host+controller on a single chip.

Commit:
Bluetooth: controller: Remove BT_HCI_RAW dependency

Removes the dependency on BT_HCI_RAW from BT_CTLR_ASSERT_HANDLER.

Signed-off-by: Thomas Stenersen <thomas.stenersen@nordicsemi.no>

cvinayak reviewed (changes requested):
Removing the dependency will cause the controller asserts to use uninitialized UART causing conflicts with application's use of UART, if any. This depend is in place in order to have HCI error event be generated when building controller only (ex, hci_uart). That said....hmmm... hci_usb is broken! controller will use UART for error!

thomasstenersen commented:
As discussed on Slack, the change in this PR does not affect UART in any way. It just allows the user the define its own assertion handler without needing to set BT_HCI_RAW when it's not used.

cvinayak commented on subsys/bluetooth/controller/Kconfig:
```diff
@@ -733,7 +733,6 @@ comment "BLE Controller debug configuration"
 
 config BT_CTLR_ASSERT_HANDLER
 	bool "Bluetooth Controller Assertion Handler"
```
Rename to "Application Defined Assertion Handler"

thomasstenersen replied:
Done.

Commit:
Bluetooth: controller: Rename assert handler Kconfig title

Renames the "Blueooth Controller Assertion Handler" to "Application
Defined Assertion Handler" to better fit the purpose of the Kconfig
option.

Signed-off-by: Thomas Stenersen <thomas.stenersen@nordicsemi.no>
